corelib/calc/crayfish_dataset_utils.cpp

```diff
@@ -130,6 +131,70 @@ void CrayfishDataSetUtils::populateSpatialFilter(DataSet& filter, const BBox& ou
     }
 }
 
+void geosNoticeFunc(const char *message, ...)
+{
+    printf("Notification from GEOS lib: %s\n", message);
+}
+
+void geosErrorFunc(const char *message, ...)
+{
+    printf("An Error occured in GEOS lib: %s\n", message);
+}
+
+void CrayfishDataSetUtils::populateMaskFilter(DataSet& filter, const QString& maskWkt) const
+{
+    filter.deleteOutputs();
+    GEOSGeometry* maskGeom;
+    GEOSContextHandle_t cx = initGEOS_r(&geosNoticeFunc, &geosErrorFunc);
+
+    GEOSWKTReader *reader = GEOSWKTReader_create_r(cx);
+    const char* mask = maskWkt.toLatin1().data();
+    maskGeom = GEOSWKTReader_read_r(cx, reader, mask);
+
+    if (mOutputType == Output::TypeNode) {
+        NodeOutput* output = new NodeOutput();
+        output->init(mMesh->nodes().size(),
+                     mMesh->elements().size(),
+                     DataSet::Scalar);
+        output->time = mTimes[0];
+        for (int i = 0; i < mMesh->nodes().size(); ++i)
+        {
+            const char* pointWkt = mMesh->projectedNode(i).toWkt().toLatin1().data();
```

Comment:
This is undefined behavior. toLatin1() creates QByteArray which is not stored and hence destructed afterwards. Hence pointWkt pointer is pointing to actually descructed object memory. Usually it works since the memory is usually not overwritten, but it is "I'm feeling lucky"...

Use instead:
```
QByteArray arr = myQString.toAscii(); 
const chat* cstr_arr = arr.constData()
```